core/src/main/java/org/apache/carbondata/core/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,171 @@
+    }
+    options = options.toLowerCase();
+    if (!options.contains(handlerName.toLowerCase())) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s is not present.",
+                      CarbonCommonConstants.INDEX_HANDLER, handlerName));
+    }
+
+    String commonKey = CarbonCommonConstants.INDEX_HANDLER + "." + handlerName + ".";
+    String TYPE = commonKey + "type";
+    String SOURCE_COLUMNS = commonKey + "sourcecolumns";
+    String SOURCE_COLUMN_TYPES = commonKey + "sourcecolumntypes";
+    String TARGET_DATA_TYPE = commonKey + "datatype";
+    String ORIGIN_LATITUDE = commonKey + "originlatitude";
+    String MIN_LONGITUDE = commonKey + "minlongitude";
+    String MAX_LONGITUDE = commonKey + "maxlongitude";
+    String MIN_LATITUDE = commonKey + "minlatitude";
+    String MAX_LATITUDE = commonKey + "maxlatitude";
+    String GRID_SIZE = commonKey + "gridsize";
+    String CONVERSION_RATIO = commonKey + "conversionratio";
+
+
+    String sourceColumnsOption = properties.get(SOURCE_COLUMNS);
+    if (sourceColumnsOption == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property is not specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+    }
+
+    if (sourceColumnsOption.split(",").length != 2) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must have 2 columns.",
+                      CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+    }
+
+    String type = properties.get(TYPE);
+    if (type != null && !CarbonCommonConstants.GEOHASH.equalsIgnoreCase(type)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be %s for this class.",
+                      CarbonCommonConstants.INDEX_HANDLER, TYPE, CarbonCommonConstants.GEOHASH));
+    }
+
+    properties.put(TYPE, CarbonCommonConstants.GEOHASH);
+
+    String sourceDataTypes = properties.get(SOURCE_COLUMN_TYPES);
+    String[] srcTypes = sourceDataTypes.split(",");
+    for (String srcdataType : srcTypes) {
+      if (!"bigint".equalsIgnoreCase(srcdataType)) {
+        throw new MalformedCarbonCommandException(
+                String.format("%s property is invalid. %s datatypes must be long.",
+                        CarbonCommonConstants.INDEX_HANDLER, SOURCE_COLUMNS));
+      }
+    }
+
+    String dataType = properties.get(TARGET_DATA_TYPE);
+    if (dataType != null && !"long".equalsIgnoreCase(dataType)) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be long for this class.",
+                      CarbonCommonConstants.INDEX_HANDLER, TARGET_DATA_TYPE));
+    }
+
+    /* Set the generated column data type as long */
+    properties.put(TARGET_DATA_TYPE, "long");
```

Comment:
If it is already there why we add again ? make it in else

Reply:
It is removed.